mirgecom/simutil.py

```diff
@@ -66,12 +66,17 @@ def inviscid_sim_timestep(discr, state, t, dt, cfl, eos,
                           t_final, constant_cfl=False):
     """Return the maximum stable dt."""
     mydt = dt
+    dt_left = t_final - t
+    if dt_left < 0:
+        return 0.0
     if constant_cfl is True:
-        mydt = get_inviscid_timestep(discr=discr, q=state,
-                                     cfl=cfl, eos=eos)
-    if (t + mydt) > t_final:
-        mydt = t_final - t
-    return mydt
+        from grudge.op import nodal_min
```

Comment:
Just a comment that `nodal_min` (and other nodal reductions like max, norm) will become distributive reductions (https://github.com/inducer/grudge/pull/117)

Reply:
This function won't work right until that lands.

Reply:
It landed haha, so `nodal_min` is now MPI-aware